logger/logger.go

```diff
@@ -47,6 +52,54 @@ func onConfigUpdated(cfg Config) {
 	}
 }
 
+type GcpFormatter struct {
+}
+
+type GcpEntry struct {
+	Data      map[string]interface{} `json:"data,omitempty"`
+	Message   string                 `json:"message,omitempty"`
+	Severity  GcpSeverity            `json:"severity,omitempty"`
+	Timestamp string                 `json:"timestamp,omitempty"`
+}
+
+type GcpSeverity = string
+
+// https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#LogSeverity
+const (
+	GcpSeverityDebug    GcpSeverity = "DEBUG"
+	GcpSeverityInfo     GcpSeverity = "INFO"
+	GcpSeverityWarning  GcpSeverity = "WARNING"
+	GcpSeverityError    GcpSeverity = "ERROR"
+	GcpSeverityCritical GcpSeverity = "CRITICAL"
+	GcpSeverityAlert    GcpSeverity = "ALERT"
+)
+
+var (
+	logrusToGcp = map[logrus.Level]GcpSeverity{
+		logrus.DebugLevel: GcpSeverityDebug,
+		logrus.InfoLevel:  GcpSeverityInfo,
+		logrus.WarnLevel:  GcpSeverityWarning,
+		logrus.ErrorLevel: GcpSeverityError,
+		logrus.FatalLevel: GcpSeverityCritical,
+		logrus.PanicLevel: GcpSeverityAlert,
+	}
+)
+
+func (f *GcpFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	var log = &GcpEntry{
+		Timestamp: entry.Time.Format(time.RFC3339),
+		Message:   entry.Message,
+		Severity:  logrusToGcp[entry.Level],
```

Comment:
You might want to make this default to Debug or something in the rare case we add a log level but not add a mapping here...

Reply:
makes sense